src/Reflector/DefaultReflector.php

```diff
@@ -56,6 +57,23 @@ public function reflectAllClasses(): iterable
         return $allClasses;
     }
 
+    /**
+     * Iterate classes available in the scope specified by the SourceLocator.
+     *
+     * @return Generator<ReflectionClass>
```

Comment:
Overall, I'd say that we can change the default `reflectAllClasses()` API to return an `iterable<ReflectionClass>`, rather than adding new API here.

Yes, it's a breaking change, but it's a good one to have